.github/workflows/pip-requirements.yml

```diff
@@ -0,0 +1,81 @@
+name: Validate pip requirements-fixed.txt
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - main
+      - 'v*-branch'
+    paths:
+      - 'scripts/requirements*.txt'
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          path: ncs/nrf
+          fetch-depth: 0
+
+      - name: Get python version
+        id: pyv
+        run: |
+          sudo snap install yq
+          PYTHON_VERSION=$(yq '.python.version' ./ncs/nrf/scripts/tools-versions-linux.yml)
+          echo "python_version=$PYTHON_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Setup python version
+        uses: actions/setup-python@v4
+        with:
+          python-version: '${{ steps.pyv.outputs.python_version }}'
+
+      - name: Setup environment
+        working-directory: ncs
+        run: |
+          pip3 install --user -U setuptools wheel pip virtualenv virtualenvwrapper
+          pip3 install -r nrf/scripts/requirements-base.txt
+          west init -l nrf
+          west update mcuboot zephyr
+
+      - name: Generate new requirements-fixed.txt
+        working-directory: ncs
+        run: nrf/scripts/compile-requirements.sh requirements-fixed.txt
+
+      - name: Store requirements-fixed
+        uses: actions/upload-artifact@v3
+        with:
+          name: requirements-fixed
+          path: ncs/requirements-fixed.txt
+
+      - name: Compare requirements-fixed
+        id: diff
+        run: |
+          NEW=ncs/requirements-fixed.txt
+          OLD=ncs/nrf/scripts/requirements-fixed.txt
+          echo -e "$NEW\t\t\t\t\t"$OLD
+          diff -y $NEW $OLD
+
+      - name: Post summary
+        if: ${{ !cancelled() }}
+        run: |
+          if [[ "failure" == "${{ steps.diff.outcome }}" ]]; then
+            echo -e 'New requirements-fixed.txt file differs from the current file.
+            Please add the newly created file to the PR.' >> $GITHUB_STEP_SUMMARY
```

Comment:
i'm aligned with @gmarull on this, ref: https://github.com/nrfconnect/sdk-nrf/pull/11994#discussion_r1304385526

The way pip packages are installed, ref: https://github.com/nrfconnect/sdk-nrf/blob/7408162be1c682b8ddcd22cebd10c14f594d835b/.github/workflows/pip-requirements.yml#L38

means that a new pip package released, which are completely independent of NCS, can cause the  requirements-fixed.txt to change.

A change completely unrelated to the PR being tested, for example, I create a PR which changes foo -> bar in main.c can result in a failed PR, because a new pip package has been released.

Fixing such failure should not be the responsibility of a PR owner (unless that PR owner has made changes to a `requirements-*` file.

If `requirements-fixed.txt` changes so frequently, as mentioned here: https://github.com/nrfconnect/sdk-nrf/pull/11994#discussion_r1306016974, then I propose to setup a daily cron job which generates `requirements-fixed.txt` and opens a PR whenever it changes.

Reply:
Fixed, see this comment: https://github.com/nrfconnect/sdk-nrf/pull/12193#issuecomment-1704855970
> Seems I've overlooked this line:
> https://github.com/nrfconnect/sdk-nrf/blob/e76e113d767bfc5a8daf52851bf79c55416be447/.github/workflows/pip-requirements.yml#L9-L10